### DISCUSSION TITLE
Bridging to IRC/Matrix

### DISCUSSION THREAD

**Discussion by div72** [vlang/v#15484] [General]
On 2022-05-19, the only channel bridged on Matrix(#vlang-random-and-offtopic) was de-bridged abruptly(abruptly as the bridge was already closed before I even saw the discussion in the same day) due to prior spam from Matrix side.

The reasoning for the cut was not unfair as the channels in Matrix do get quite a bit of spam. But in the meantime the channels got an additional moderator(me) and a anti-spam bot([mjolnir](https://github.com/matrix-org/mjolnir) instance) that uses banlists(lists which are updated by other channels so that spammers blocked because of spam from a server will be blocked on other servers) and wordlists(limitations on content posted on arrival) in order to moderate the rooms.

I would like to ask for a re-trial of the Matrix bridge(at least in offtopic channels) since the spam is next to none nowadays. In case the spam increases after bridging, I propose not to cut off the bridge directly but make the rooms private and allow Matrix users to be vetted in a separate room before they are allowed access to the other rooms.

In case that is somehow unacceptable, I propose that V should have an IRC bridge on libera.chat instead. Since the time I started using libera, I have never seen spam in the rooms I am in.

## Why?

Simple, V is an open source language/compiler. It deserves to be on an open platform. Some might argue that the other platforms do not have any users. I'd like to argue that that is also caused by the channels not being bridged. As an example, there were some users who joined asking for help and leaving as the channels were empty.

Related: https://drewdevault.com/2021/12/28/Dont-use-Discord-for-FOSS.html

@medvednikov

**Comment by medvednikov** [comment]
We were getting lots of spam from Matrix.

**Reply by div72** [reply]
I know. The channels have an anti-spam bot that deals with the spam now.

**Reply by medvednikov** [reply]
Ok I can set it up again and see how it goes.

**Comment by spytheman** [comment]
From the article:
> It sets up users who are passionate about free software — i.e. your most passionate contributors or potential contributors — as second-class citizens.

That is a false dichotomy. For example, I am passionate about free software, but I am not a fanatic/purist, and since I do consider Discord superior to other chat systems -> I use it, even though it is not free software.

The accessibility exclusion point there is also pure BS - you can use Discord from a web browser, so it is as accessible/cross platform as any web app.

The conclusion from that article:
>> Perceptive readers might have noticed that most of these arguments can be generalized. This article is much the same if we replace “Discord” with “GitHub”, for instance, or “Twitter” or “YouTube”. If your project depends on proprietary infrastructure, I want you to have a serious discussion with your collaborators about why.

The reason is very simple - because of network effects, and because they are much more convenient for most users. These proprietary platforms have attracted a critical mass of users, because they were subsidized and developed more convenience features, while the open source platforms were either non existent, or were very buggy and unusable.

I think that all opinions of Mr. DeVault, about open source hosting and services have to be taken with a pinch of salt, given that afaik he develops Sourcehut, i.e. he may not be very objective on this topic.
